### DISCUSSION TITLE
[OSPP Week 4] Add d4rl-pybullet datasets and RLUnplugged datasets in RLDatasets.jl

### DISCUSSION THREAD

**Discussion by Mobius1D** [JuliaReinforcementLearning/ReinforcementLearning.jl#413] [Archive]
So far the work done is:

Thanks to @findmyway for his patience in explaining various concepts and insights into design I was able to put out a basic implementation of RLDatasets.jl. 

Release the basic version RLDatasets.jl enabling the use of d4rl datasets out of the box and make it convenient for use in Offline Reinforcement Learning by returning an iterator.

The work for this week would be:

- [x] Extend the functionality to include [d4rl-pybullet](https://github.com/takuseno/d4rl-pybullet).
  - This will enable the user to benchmark multi action continuous control agents without access to a MuJoCo licence.
- [x] Add https://github.com/google-research/batch_rl datasets for atari envs.
- [x] Add RLUnplugged datasets for standard bench-marking in atari envs.
- [x] Improve the documentation for ReinforcementLearningDatasets to make it user friendly.
- [ ] Design an interface for user to collect, store and retrieve datasets so that users can use RLDatasets.jl using their own datasets seamlessly. The retrieved datasets can be an iterator. @findmyway Please give your thoughts on this.
- [x] Understand the basics of multi threaded experience collection and open a repository (in my account) to document my learning.
- [x] Set up the docs.

This would amount to a majority of work that would be needed to register RLDatasets.jl. 

@pilgrimygy Please give your thoughts on the features that you would need to make this package usable for offline training. The features for storing and retrieving datasets will be implemented soon.

@findmyway It would be great if you could give your thoughts on the plan.

I also think that we would be needing a storage facility to put the datasets that we would be releasing down the line.

**Comment by findmyway** [comment]
Also set up the docs?

**Reply by Mobius1D** [reply]
Hmm. Yes
